Improving description display on Evolution Dashboard; Fixes #358

shikhar-scs opened the pull request:
As widely discussed at #358, I have made a few changes to improve the appearance of the description dashboard.

Here's a Priview

![evodash](https://user-images.githubusercontent.com/25258877/34742839-a8ed33ea-f5ad-11e7-9f24-c35825d4d699.gif)


Please review @georgf

shikhar-scs committed:
removed prism files

shikhar-scs committed:
improved description display on evolution dashboard

shikhar-scs commented on lib/prism.css:
```diff
@@ -1,167 +0,0 @@
-		}
-.token a {
-	color: inherit;
-}
```
Please ignore this commit, it references #367. I shall rebase them in my next push. Also, please ignore prism.js and package-lock.json

shikhar-scs replied:
Please ignore this file, it actually references #367 which has already been merged. Similarly please ignore prism.js and package,json

shikhar-scs commented:
@georgf do u mind reviewing here and suggesting some changes ?

georgf commented:
Cheers for the screenshot, that makes it easy to give feedback.

As mentioned in #358, we want:
- the description to not be a link, just plain text style (it looks like it's still a link)
- add an an extra link or button ("More details"), that links to the same URL
Looking at [max_concurrent_tabs](https://mzl.la/2ASwYc8) probably makes it more clear why we want that.

The arrow animation looks nice, but i don't think it's needed here.

shikhar-scs committed:
# This is a combination of 5 commits.
# The first commit's message is:
added more-details button, removed link from heading

# This is the 2nd commit message:

conflicting prism.css file

# This is the 3rd commit message:

incorporated required changes

# This is the 4th commit message:

made changes

# This is the 5th commit message:

made changes

shikhar-scs commented:
@georgf I've pushed new changes and they look much better now. Here's a preview.
![evo-dashboard](https://user-images.githubusercontent.com/25258877/34904127-20e1992c-f865-11e7-81aa-d84b1d4f1720.png)


![evo-dsas](https://user-images.githubusercontent.com/25258877/34904128-229f7676-f865-11e7-83f5-4c9694603ad7.png)

These changes are responsive too.
Please review

shikhar-scs commented on lib/prism.css:
```diff
@@ -98,70 +98,70 @@ pre[class*="language-"] {
 .token a {
-	color: inherit;
+        color: inherit;
 }
```
There  have been some useless conflicting errors. I had initially filed a PR #367 (merged already) which had already deleted these unused `prism.js` and `prism.css` files. So merging them won't be a problem .

georgf replied:
Can you get update your fork from the current master and get rid of the extra changes here?

georgf commented:
This looks better, cheers.
Some things:
- It looks both description & button are all capitalized now, let's not do that.
- This looks like it uses a different font? Let's just use the current one. (We're aiming for a minimal change here.)
- Let's make the button black or grey. Visually the dashboard is already pretty busy, the blue button distracts a bit from the main visualization below.

shikhar-scs commented:
> It looks both description & button are all capitalized now, let's not do that.

Actually all stuff in the description tag is passed through `.toupper` I guess. I had myself made it in lowercase only. I'll try and make changes again and report back

> This looks like it uses a different font? Let's just use the current one. (We're aiming for a minimal change here.)

Its just bigger in size maybe, I'll restore it back.

> Let's make the button black or grey. Visually the dashboard is already pretty busy, the blue button distracts a bit from the main visualization below.

Sure :+1:

shikhar-scs commented on new-pipeline/src/dashboards.js:
```diff
@@ -1048,8 +1048,7 @@ function updateOSs() {
 
-  return " <a href=\"" + metricUrl + "\" target=\"_blank\">" + descr + "<img src=\"" + imgUrl + "\"/></a>"
+  return descr + " <a href=\"" + metricUrl + "\" target=\"_blank\">" + "<i id=\"more-details\" class=\"btn btn-outline-primary fa fa-info-circle\" style='float: right' aria-hidden=\"true\">  more details </i></a>"
 }
```
see here I've used `more details` in the lowercase

shikhar-scs commented:
@georgf 

I've introduced some changes, please have a look, the tag is black now and the original font has been restored.

I'm still unable to find the exact place where the equivalent of `description.toUppercase()` happens. Could you guide me on this please.

We can also do away with the `more details` tag, just keeping the icon in place, whatsay ?


![heeeeyyy](https://user-images.githubusercontent.com/25258877/34919212-83040ab0-f985-11e7-8187-37bd2c2c394b.png)

georgf commented on new-pipeline/evo.html:
```diff
@@ -62,7 +62,7 @@ <h2 style="font-size: 18px; margin-bottom: 0">
                     <select id="filter-process-type" class="multiselect" title="Process type" data-all-selected="any process type"></select>
                 </h2>
-                <h3 id="measure-description" style="font-size: 14px; margin: 0 3px"></h3>
+                <h3 id="measure-description" style="font-size: 20px; margin: 0 3px"></h3>
```
The changes in this file look unneeded?

georgf commented on new-pipeline/style/custom.css:
```diff
@@ -0,0 +1,75 @@
+.button {
```
Let's find existing bootstrap styles that we can use instead of adding custom CSS.
The text-transform in here is what makes the whole text upper-case.

shikhar-scs replied:
> The text-transform in here is what makes the whole text upper-case.

Thank you for pointing that out.

> Let's find existing bootstrap styles that we can use instead of adding custom CSS.

That would be difficult I guess, given the amount of css it is. However, if that's really needed, I shall work on it.

shikhar-scs replied:
@georgf Can I try and make it as least as possible maybe ?

georgf replied:
You don't need to go through all the CSS.
The project uses Bootstrap v2.0.3, the documentation for that can be found here:
https://getbootstrap.com/2.0.3/index.html

georgf commented on new-pipeline/src/evo.js:
```diff
@@ -337,6 +337,13 @@ $(function () {
               var metric = $("#measure").val()
               var channel = $("#max-channel-version").val().split("/")[0]
               var description = getDescriptionWithLink(metric, channel, evolutionDescription)
+              var improvedDescription = document.createElement("button");
```
Let's just keep the changes inside `getDescriptionWithLink()` in dashboards.js for now.

shikhar-scs replied:
sure :+1: I'll transfer all of this there only

chutten commented:
The text isn't uppercase in the live version at telemetry.mozilla.org. Maybe there's a `text-transform` someplace in your added CSS?

shikhar-scs commented on lib/jquery.js:
```diff
@@ -92,7 +92,7 @@ var
 	// Used by jQuery.camelCase as callback to replace()
 	fcamelCase = function( all, letter ) {
-		return letter.toUpperCase();
+		return letter;
```
I'll change this back

shikhar-scs commented:
@georgf I have completely removed the custom.css file and realized that we don't need that button thing in place at all ... Please review all the other changes

georgf commented on new-pipeline/src/dashboards.js:
```diff
@@ -1048,8 +1048,12 @@ function updateOSs() {
-  return " <a href=\"" + metricUrl + "\" target=\"_blank\">" + descr + "<img src=\"" + imgUrl + "\"/></a>"
+  var div = document.createElement('div');
+  div.classList.add("text-center");
+  div.classList.add("py-5");
```
I don't think these two classes are needed, let's remove them.

shikhar-scs replied:
We do need `text-center` otherwise the descriptions come to the left.
I'll remove `py-5` for sure.

georgf commented on new-pipeline/src/dashboards.js:
```diff
@@ -1048,8 +1048,12 @@ function updateOSs() {
+  var div = document.createElement('div');
+  div.classList.add("text-center");
+  div.classList.add("py-5");
+  div.innerHTML =  descr + " <a href=\"" + metricUrl + "\" target=\"_blank\">" + "<i class=\"btn btn-outline-primary fa fa-info-circle\" style='float: right; color: black' aria-hidden=\"true\">  more details </i></a>"
```
Visually this looks pretty close.
Some things:
- Use "More details" vs. "more details"
- I don't think the separate `<i>` element is needed, you can just style the `<a>` element.
- This line is pretty long, let's break it up.
- Let's drop the `float: right`, it looks off and makes the button hard to find.

shikhar-scs replied:
`<i>` element is needed for the icon `fa fa-info-circle` of `More Details` otherwise if you don't need the icon I'll remove it from there :) 

`float:right` helps to keep the `More Details` part to the right. Else it comes just after the description which doesn't look good.

This 
![1](https://user-images.githubusercontent.com/25258877/34958861-b92fc438-fa59-11e7-9c22-eec03d50fde1.png)
vs this
![2](https://user-images.githubusercontent.com/25258877/34958864-bf65ffca-fa59-11e7-8d4d-1434fd7e4de8.png)

shikhar-scs commented:
Please have a look, also I'll rebase my commits once this is ready to merge.

himanish-star committed:
rebased

georgf commented on new-pipeline/src/dashboards.js:
```diff
@@ -1048,8 +1048,14 @@ function updateOSs() {
+  div.innerHTML =  descr + " <a href=\"" + metricUrl + "\" target=\"_blank\">" +
+                   "<i class=\"btn btn-outline-primary fa fa-info-circle\" " +
+                   "style='float: right; color: black' aria-hidden=\"true\">  " +
+                   "More Details </i></a>"
```
Two things here from the previous feedback that were not addressed:
- This still uses the `<i>` element.
- This still uses `float: right;`

shikhar-scs replied:
I had already mentioned previously that the icon of more details uses the `<i>` tag so only if you don't want the icon I shall remove it. So 

I had posted screenshots related to why the absence of `float : right` spoils the view , basically because then `more details` just sticks right next to where the description ends . However I shall remove that if that is what is really needed.

shikhar-scs committed:
merge conflicts

shikhar-scs committed:
removed float:right

shikhar-scs commented:
@georgf Any more changes you need ?

georgf commented on new-pipeline/src/dashboards.js:
```diff
@@ -204,7 +204,7 @@ function loadStateFromUrlAndCookie() {
     pageState.os = pageState.arch = pageState.e10s = pageState.processType =
       null;
-    pageState.compare = "";
+    pageState.compare ="";
```
Please drop this change.

shikhar-scs replied:
sure

georgf commented on new-pipeline/src/dist.js:
```diff
@@ -3,7 +3,7 @@ var gFilterChangeTimeout = null;
 var gCurrentDates = null;
 var gCurrentMinDate = null,
-  gCurrentMaxDate = null;
+    gCurrentMaxDate = null;
```
There are a lot of whitespace changes in this file, that we don't need to change.
I get a syntax error in this file too when testing the changes locally.
Please drop the changes to this file.

georgf commented on new-pipeline/evo.html:
```diff
@@ -4,17 +4,17 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    
+
```
There are whitespace changes in this file that we don't need.
Please drop these changes.

shikhar-scs replied:
Yeah I will.

georgf commented on new-pipeline/src/evo.js:
```diff
@@ -26,7 +26,7 @@ var gDefaultAggregates = [
 ];
 // these will be generated, but won't appear in the multiselect
-// note: some code using gMetaAggregates assumes for convenience that 
+// note: some code using gMetaAggregates assumes for convenience that
```
There are whitespace changes in this file that we don't need.
Please drop these changes.

shikhar-scs replied:
Yep. Actually all these changes are occurring because my editor has a different default indentation. And so if I open a file by mistake, it gets altered.

I'll checkout these changes

georgf commented on new-pipeline/src/dashboards.js:
```diff
@@ -1048,8 +1048,14 @@ function updateOSs() {
 
 function getDescriptionWithLink(metric, channel, description) {
   var metricUrl = "https://georgf.github.io/fx-data-explorer/index.html?search=" + metric + "&searchtype=in_name&optout=false&channel=" + channel + "&constraint=is_in&version=any"
-  var imgUrl = "https://upload.wikimedia.org/wikipedia/commons/6/64/Icon_External_Link.png"
```
This was the only use of this file, can you remove the actual file too?

shikhar-scs replied:
yeah sure

georgf commented on new-pipeline/lib/jquery.js:
```diff
@@ -92,7 +92,7 @@ var
 	// Used by jQuery.camelCase as callback to replace()
 	fcamelCase = function( all, letter ) {
-		return letter.toUpperCase();
+		return letter;
```
Please drop this change.

shikhar-scs replied:
sure

georgf reviewed (changes requested):
Thanks, functionally this seems to be where we want it.
There are some things that still need cleaning up though.

shikhar-scs committed:
ready for merge

georgf commented:
Looking [over here](https://github.com/mozilla/telemetry-dashboard/pull/368/files), not all of the requested changes happened yet.
There is also an expiry change in dist.js that we need to drop.

shikhar-scs commented:
Due to some poor rebasing habits, I've actually messed up the code a bit.
Can I close this and file another PR for the same please?

georgf commented:
You could also clean up the remaining unneeded changes, there are not many left!
But if a new PR works easier for you feel free to.

shikhar-scs commented:
Yeah sure @georgf

shikhar-scs commented:
Closed in favour of #380
